File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        best = 0

        for n in nums:
            if n-1 not in nums:
                length = 1
                nxt = n + 1

                while nxt in nums:
                    length += 1
                    nxt += 1
                
                best = max(best, length)

        return best

File: Data Structures & Algorithms/house-robber/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        const n = nums.length;
        
        if (n == 0) return 0;
        if (n <= 2) return Math.max(...nums);

        let prev2 = nums[0];
        let prev1 = Math.max(nums[0], nums[1]);

        for (let i = 2; i < n; i++) {
            let curr = Math.max(prev1, prev2 + nums[i]);

            prev2 = prev1;
            prev1 = curr;
        }

        return prev1;
    }
}
